analyzers/src/SonarAnalyzer.VisualBasic/Rules/Naming/TypeParameterName.cs

```diff
@@ -23,34 +23,33 @@ namespace SonarAnalyzer.Rules.VisualBasic
     [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
     public sealed class TypeParameterName : ParametrizedDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S2373";
-        private const string MessageFormat = "Rename '{0}' to match the regular expression: '{1}'.";
-
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat,
-                isEnabledByDefault: false);
+        private const string S119DiagnosticId = "S119";
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        [Obsolete("This rule is superseded by S119.")]
+        private const string S2373DiagnosticId = "S2373";
 
+        private const string MessageFormat = "Rename '{0}' to match the regular expression: '{1}'.";
         private const string DefaultFormat = "^T(" + NamingHelper.PascalCasingInternalPattern + ")?";
 
-        [RuleParameter("format", PropertyType.String,
-            "Regular expression used to check the generic type parameter names against.", DefaultFormat)]
+        internal static readonly DiagnosticDescriptor S119 = DescriptorFactory.Create(S119DiagnosticId, MessageFormat, isEnabledByDefault: false);
+        internal static readonly DiagnosticDescriptor S2373 = DescriptorFactory.Create(S2373DiagnosticId, MessageFormat, isEnabledByDefault: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(S119, S2373);
+
+        [RuleParameter("format", PropertyType.String, "Regular expression used to check the generic type parameter names against.", DefaultFormat)]
         public string Pattern { get; set; } = DefaultFormat;
 
-        protected override void Initialize(SonarParametrizedAnalysisContext context)
-        {
+        protected override void Initialize(SonarParametrizedAnalysisContext context) =>
             context.RegisterNodeAction(
                 c =>
                 {
                     var typeParameter = (TypeParameterSyntax)c.Node;
                     if (!NamingHelper.IsRegexMatch(typeParameter.Identifier.ValueText, Pattern))
                     {
-                        c.ReportIssue(Diagnostic.Create(rule, typeParameter.Identifier.GetLocation(),
-                            typeParameter.Identifier.ValueText, Pattern));
+                        c.ReportIssue(Diagnostic.Create(S119, typeParameter.Identifier.GetLocation(), typeParameter.Identifier.ValueText, Pattern));
+                        c.ReportIssue(Diagnostic.Create(S2373, typeParameter.Identifier.GetLocation(), typeParameter.Identifier.ValueText, Pattern));
```

Comment:
NITPICK:
Instead of retrieveing the valueText and the location twice, save them in a variable and forward it to both invocations.